backend-rust/src/migrations/m0019_payday_stake_information.rs

```diff
@@ -37,26 +41,38 @@ pub async fn run(
     tx: &mut Transaction,
     endpoints: &[v2::Endpoint],
 ) -> anyhow::Result<SchemaVersion> {
-    tx.as_mut()
-        .execute(sqlx::raw_sql(include_str!("./m0018-payday-stake-information.sql")))
-        .await?;
-
-    let endpoint = endpoints.first().with_context(|| {
-        format!("Migration '{}' must be provided access to a Concordium node", NEXT_SCHEMA_VERSION)
-    })?;
-    let mut client = v2::Client::new(endpoint.clone()).await?;
-
+    let last_processed_payday: Option<i64> =
+        sqlx::query_scalar("SELECT MAX(payday_block) FROM payday_baker_pool_stakes")
+            .fetch_one(tx.as_mut())
+            .await?;
+    // Fetch the paydays that we have not processed yet.
     let paydays: Vec<i64> = sqlx::query_scalar(
         "SELECT DISTINCT payday_block_height
          FROM bakers_payday_pool_rewards
+         WHERE payday_block_height > $1
          ORDER BY payday_block_height ASC",
     )
+    .bind(last_processed_payday.unwrap_or(i64::MIN))
     .fetch_all(tx.as_mut())
     .await?;
+    // If no more paydays to process we are done migrating.
+    if paydays.is_empty() {
```

Comment:
Mentally I find a termination condition being placed in the middle of a script being pretty annoying.

That said this does not have to be long lived or anything.

An alternative way could be to treat paydays as a queue and then make this check in the end.

Reply:
we do treat it as a queue.
It needs to be placed here, to ensure we only migrate to the next schema version when we start this migration with no paydays to process.